### PR TITLE
Wrapper for ShoRAH version 1.1.3 in amplicon mode

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/shorah/shorah.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    <description>with ShoRAH in amplicon mode</description>
+    <macros>
+        <token name="@VERSION@">1.1.3</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">shorah</requirement>
+    </requirements>
+    <command detect_errors="exit_code">
+    <![CDATA[
+    ln -s '$bam' input.bam &&
```
Comment (bgruening) [inline tools/shorah/shorah.xml]: no index needed?

File: tools/shorah/test-data/shorah-amplicon-out1.csv [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Chromosome,Pos,Ref,Var,Freq,Post,Fvar,Rvar,Ftot,Rtot,Pval,Qval
```
Comment (bgruening) [inline tools/shorah/test-data/shorah-amplicon-out1.csv]: can we convert this to tabular inside the tool already?

File: tools/shorah/shorah.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        <param argument="--region" type="text" value="" optional="true" label="Limit to a specific region" help="e.g. 'ch3:1000-1300'" />
+        <param argument="--diversity" type="boolean" truevalue="--diversity" falsevalue="" label="Run on the highest entropy region" />
+        <param argument="--min_overlap" type="float" value="0.95" min="0" max="1" optional="true" label="Fraction of read overlap to be included" />
+        <param argument="--alpha" type="float" value="0.5" optional="true" label="Alpha in dpm sampling" />
+        <param argument="--maxcov" type="integer" value="50000" optional="true" label="Approximate max coverage allowed" />
+        <param argument="--sigma" type="float" value="0.01" optional="true" label="Sigma value to use when calling SNVs" />
+    </inputs>
+    <outputs>
+        <data name="haplotypes" format="csv" from_work_dir="SNVs_0.010000_final.csv" label="${tool.name} on ${on_string}: Haplotypes" />
+        <data name="log" format="txt" from_work_dir="SNV.txt" label="${tool.name} on ${on_string}: Log" />
```
Comment (bgruening) [inline tools/shorah/shorah.xml]: Should that be optional?

File: tools/shorah/shorah.xml (hunk 1 above)
Comment (davebx) [reply tools/shorah/shorah.xml]: It seems to function without it, though it does generate one.

Comment (bgruening) [reply tools/shorah/shorah.xml]: So if you provide it with one, maybe it does not create one?

**Comment by bgruening** [comment]
ping @davebx failing here ...

**Comment by davebx** [comment]
Ping @bgruening no it's not :grin:

**Comment by bgruening** [comment]
Thanks @davebx!

**Pull request merged**